### DISCUSSION TITLE
Proposal to add generic LoRa pin names

### DISCUSSION THREAD

**Discussion by Jaimi5** [espressif/arduino-esp32#7946] [Ideas]
Hey fellow developers!

I'd like to propose an idea that's related to LoRa devices. As you may know, the framework currently has different files for each compatible device, and each file has specific pin definitions. However, some devices have different pin configurations, which can make it difficult to write portable code across devices.

So, my proposal is to add generic LoRa pin names that can be used across different devices. This way, we could write portable code more easily without worrying about device-specific pin configurations.

I've noticed that some devices already have these generic LoRa pin names, for example, the TTGO devices have the following pin names for lora:  LORA_SCK, LORA_MISO, LORA_MOSI, LORA_CS, LORA_RST, LORA_IRQ. 
But not all of them do. So, I think we should extend these names to other devices to have a consistent set of names across all devices.

Let me know your thoughts on this proposal. If it is okey I would add a pull request to add these pins for some of the devices that uses LoRa.

Thanks a lot!

**Comment by mrengineer7777** [comment]
Are you proposing adding LORA definitions in `pins_arduino.h`, like https://github.com/espressif/arduino-esp32/blob/master/variants/ttgo-lora32-v1/pins_arduino.h?

Sounds like a good idea as long as it matches the hardware spec for that board.  @VojtechBartoska What do you think?

According to `boards.txt` the following boards reference LORA in their name or description:
```
ttgo-lora32.name=TTGO LoRa32-OLED
   .build.board=TTGO_LORA32
sparkfun_lora_gateway_1-channel.name=SparkFun LoRa Gateway 1-Channel
   .build.board=ESP32_DEV   Should this reference here: https://github.com/espressif/arduino-esp32/blob/master/variants/sparkfun_lora_gateway_1-channel/pins_arduino.h?
heltec_wifi_lora_32.name=Heltec WiFi LoRa 32
   .build.board=HELTEC_WIFI_LORA_32
heltec_wifi_lora_32_V2.name=Heltec WiFi LoRa 32(V2)
   .build.board=HELTEC_WIFI_LORA_32_V2
heltec_wifi_lora_32_V3.name=Heltec WiFi LoRa 32(V3) / Wireless shell(V3) / Wireless stick lite (V3)
   .build.board=HELTEC_WIFI_32_LORA_V3
heltec_wireless_stick_lite.name=Heltec Wireless Stick Lite
   .build.board=HELTEC_WIRELESS_STICK_LITE
```

**Reply by Jaimi5** [reply]
Exactly!
All of these devices would benefit from adding pins. 
Should I start making the pull request with these changes?

**Reply by me-no-dev** [reply]
There is a plan for such thing not only for Lora, but for on-board screens, sensors, etc. Stay tuned :)

**Reply by Jaimi5** [reply]
Great news! I'll definitely stay tuned for updates. I'd love to attend the next meeting and learn more about these developments. If there's any way I can assist in the process, please don't hesitate to let me know. I would be more than happy to help out.
